api/repositories/app_repository.go

```diff
@@ -408,6 +415,29 @@ func (f *AppRepo) CreateOrPatchAppEnvVars(ctx context.Context, authInfo authoriz
 	return appEnvVarsSecretToRecord(secretObj), nil
 }
 
+func (f *AppRepo) PatchAppMetadata(ctx context.Context, authInfo authorization.Info, message PatchAppMetadataMessage) (AppRecord, error) {
+	userClient, err := f.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return AppRecord{}, fmt.Errorf("failed to build user client: %w", err)
+	}
+
+	app := new(korifiv1alpha1.CFApp)
+	err = userClient.Get(ctx, client.ObjectKey{Namespace: message.SpaceGUID, Name: message.AppGUID}, app)
+	if err != nil {
+		return AppRecord{}, fmt.Errorf("failed to get app: %w", apierrors.FromK8sError(err, AppResourceType))
+	}
+
+	origApp := app.DeepCopy()
+	patchMap(app.Labels, message.Labels)
+	patchMap(app.Annotations, message.Annotations)
+	err = userClient.Patch(ctx, app, client.MergeFrom(origApp))
+	if err != nil {
+		return AppRecord{}, apierrors.FromK8sErrorWithInvalidAsUnprocessableEntity(err, AppResourceType)
```

Comment:
I think this is fair to do. Feels better than try to copy over Cloud Controller's approximation of the Kubernetes labels/annotations validation logic:

https://github.com/cloudfoundry/cloud_controller_ng/blob/2ab201a7c53716cff7c23c35f1cb33ed34f8c517/app/messages/metadata_validator_helper.rb